### ISSUE TITLE
The index of Unknown used in //ModelStructure/Outputs is wrong

### ISSUE THREAD

**Issue by lausdahl** [controllab/fmi-export-20sim#1]
The index used in the index attribute in Unknown in outputs under model structure is wrong:

``` xml
<?xml version="1.0" encoding="ISO-8859-1"?>
<fmiModelDescription fmiVersion="2.0" modelName="tank2" guid="{67d835d6-8a93-41f6-bd55-ed668d3c5b52}" generationTool="20-sim" numberOfEventIndicators="0" copyright="Controllab Products B.V." license="-">
<CoSimulation modelIdentifier="tank2" needsExecutionTool="false" canHandleVariableCommunicationStepSize="false" canInterpolateInputs="true" maxOutputDerivativeOrder="0" canRunAsynchronuously="false" canBeInstantiatedOnlyOncePerProcess="true" canNotUseMemoryManagementFunctions="true" canGetAndSetFMUstate="false" canSerializeFMUstate="false" providesDirectionalDerivative="false" />
<DefaultExperiment startTime="0.0" stopTime="16.0" />
<ModelVariables>
<ScalarVariable name="Drain1.p.p" valueReference="8" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="Drain1.r" valueReference="1" variability="fixed" causality="parameter">
<Real start="1.0" />
</ScalarVariable>
<ScalarVariable name="Drain1.puddle" valueReference="9" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="tank2.Node1.p1.p" valueReference="10" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="tank2.Node1.p2.p" valueReference="10" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="tank2.Node1.p3.p" valueReference="10" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="tank2.Node1.p1.phi" valueReference="14" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="tank2.Node1.p2.phi" valueReference="24" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="tank2.Node1.p3.phi" valueReference="12" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="tank2.Tank.p.e" valueReference="10" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="tank2.Tank.level" valueReference="11" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="tank2.Tank.area" valueReference="2" variability="fixed" causality="parameter">
<Real start="1.0" />
</ScalarVariable>
<ScalarVariable name="tank2.Tank.gravity" valueReference="3" variability="fixed" causality="parameter">
<Real start="9.81" />
</ScalarVariable>
<ScalarVariable name="tank2.Tank.liquid_density" valueReference="4" variability="fixed" causality="parameter">
<Real start="1.0" />
</ScalarVariable>
<ScalarVariable name="tank2.p1.p" valueReference="10" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="tank2.p1.phi" valueReference="14" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="tank2.p.p" valueReference="10" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="tank2.p.phi" valueReference="12" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="tank2.level" valueReference="11" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="valve2.FlowOut.p.e" valueReference="8" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="valve2.FlowOut.p.f" valueReference="23" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="valve2.FlowOut.flow" valueReference="23" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="valve2.FlowOut.effort" valueReference="8" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="valve2.R.p.e" valueReference="15" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="valve2.R.p.f" valueReference="23" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="valve2.R.FlowOut" valueReference="23" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="valve2.R.r" valueReference="5" variability="fixed" causality="parameter">
<Real start="7.0" />
</ScalarVariable>
<ScalarVariable name="valve2.R.waterFlow" valueReference="13" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="valve2.Valve.valvecontrol" valueReference="16" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="valve2.Valve.powerIn.p" valueReference="10" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="valve2.Valve.powerIn.phi" valueReference="14" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="valve2.Valve.powerOut.p" valueReference="15" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="valve2.Valve.powerOut.phi" valueReference="23" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="valve2.waterOut.e" valueReference="8" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="valve2.waterOut.f" valueReference="23" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="valve2.valveControl" valueReference="16" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="valve2.waterIn.p" valueReference="10" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="valve2.waterIn.phi" valueReference="14" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="p.p" valueReference="17" variability="continuous" causality="output">
<Real />
</ScalarVariable>
<ScalarVariable name="p.phi" valueReference="18" variability="continuous" causality="input">
<Real start="0.0" />
</ScalarVariable>
<ScalarVariable name="level" valueReference="19" variability="continuous" causality="output">
<Real />
</ScalarVariable>
<ScalarVariable name="valveControl" valueReference="20" variability="discrete" causality="input">
<Boolean start="0" />
</ScalarVariable>
<ScalarVariable name="Drain1.p.phi" valueReference="23" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="Drain1.p.phi_int" valueReference="21" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="Drain1.p.phi_int_initial" valueReference="6" variability="fixed" causality="parameter">
<Real start="0.0" />
</ScalarVariable>
<ScalarVariable name="tank2.Tank.p.f" valueReference="24" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="tank2.Tank.volume" valueReference="22" variability="continuous" causality="local">
<Real />
</ScalarVariable>
<ScalarVariable name="tank2.Tank.volume_initial" valueReference="7" variability="fixed" causality="parameter">
<Real start="0.0" />
</ScalarVariable>
</ModelVariables>
<ModelStructure>
<Outputs>
<Unknown index="17" />
<Unknown index="19" />
</Outputs>
</ModelStructure>
</fmiModelDescription>
```

it should have been 

``` xml
    <ModelStructure>
        <Outputs>
            <Unknown index="39" />
            <Unknown index="41" />
        </Outputs>
    </ModelStructure>
```

Note that `index` is the index of the scalarvariable = the number in the list of declared

**Comment by MaartenMJR** [comment]
Hi Kenneth,

Thank you for reporting this bug. I've fixed this in the following commit: https://github.com/controllab/fmi-export-20sim/commit/7a908a2524794b3398a2c367a00a2c855fa6ff91

**Comment by margro** [comment]
Should be fixed now with 5c0af366e59102e20f0b6340784e067a9229f2e0
